src/gui/widgets/MicrotunerConfig.cpp

```diff
@@ -0,0 +1,650 @@
+/*
+ * MicrotunerConfig.cpp - configuration widget for scales and keymaps
+ *
+ * Copyright (c) 2020 Martin Pavelek <he29.HS/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "MicrotunerConfig.h"
+
+#include <QFile>
+#include <QGridLayout>
+#include <QLabel>
+#include <QMessageBox>
+#include <QPushButton>
+#include <QRegExp>
+#include <QTextStream>
+
+#include "ComboBox.h"
+#include "embed.h"
+#include "Engine.h"
+#include "FileDialog.h"
+#include "GuiApplication.h"
+#include "Knob.h"
+#include "LcdSpinBox.h"
+#include "lmms_constants.h"
+#include "lmmsversion.h"
+#include "MainWindow.h"
+#include "Song.h"
+
+
+MicrotunerConfig::MicrotunerConfig() :
+	QWidget(),
+	m_scaleComboModel(NULL, tr("Selected scale")),
+	m_keymapComboModel(NULL, tr("Selected keymap")),
```

Comment:
We're on a trek to start replacing `NULL` with `nullptr`. Could you test replacing all?